### PR TITLE
Implement committee_hash lightclient api

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

- Kevlar https://github.com/lightclients/kevlar needs this for its optimistic sync

**Description**

- From a period, get best update
- Extract attester header than beacon block root from there
- Get sync committee witness from `syncCommitteeWitness` db
- Get current sync committee from `syncCommittee` db

part of #4733

### CODE REVIEW

**Comment by shresthagrawal** [comment]
Thanks for taking the task! I guess you haven't added the hashing part yet? The hash should be simply the keccak256 of the concatenated public keys of the sync committee. [Here](https://github.com/lightclients/kevlar/blob/main/src/provers/light-optimistic/memory-store.ts#L20) is the current prover implementation

File: packages/api/src/beacon/routes/lightclient.ts [hunk 1]
```diff
@@ -60,6 +64,7 @@ export const routesData: RoutesData<Api> = {
   getOptimisticUpdate: {url: "/eth/v1/beacon/light_client/optimistic_update", method: "GET"},
   getFinalityUpdate: {url: "/eth/v1/beacon/light_client/finality_update", method: "GET"},
   getBootstrap: {url: "/eth/v1/beacon/light_client/bootstrap/{block_root}", method: "GET"},
+  getCommitteeHash: {url: "/eth/v1/beacon/light_client/committee_hash", method: "GET"},
```
Comment (dapplion) [inline packages/api/src/beacon/routes/lightclient.ts]: Use `v0` until this route is spec'ed in beacon-APIs repo

**Review by dapplion — DISMISSED** [review]

File: packages/beacon-node/src/chain/lightClient/index.ts [hunk 2]
```diff
@@ -291,6 +293,31 @@ export class LightClientServer {
     return update;
   }
 
+  /**
+   * API ROUTE to get the current sync committee from the best available update for `period`.
```
Comment (dadepo) [inline packages/beacon-node/src/chain/lightClient/index.ts]: minor: the comment here is semantically different from the corresponding comment on api route

**Comment by dadepo** [comment]
Maybe consider adding tests that covers the added endpoint? apart from that LGTM.

**Comment by twoeths** [comment]
> Maybe consider adding tests that covers the added endpoint? apart from that LGTM.

it already covered the end point as long as we update the testData for lightclient, doubled checked it via the unit test output

```
  beacon / lightclient
onError: Error: aborted
    ✔ getStateProof
    ✔ getUpdates
    ✔ getOptimisticUpdate
    ✔ getFinalityUpdate
    ✔ getBootstrap
    ✔ getCommitteeHash
```

**Comment by dapplion** [comment]
> Thanks for taking the task! I guess you haven't added the hashing part yet? The hash should be simply the keccak256 of the concatenated public keys of the sync committee. [Here](https://github.com/lightclients/kevlar/blob/main/src/provers/light-optimistic/memory-store.ts#L20) is the current prover implementation

@shresthagrawal Why don't you do the hashTreeRoot of the SyncCommittee instead of concat pubkeys + hash? It's kindof tangent to your classic object hashing in the beacon chain. Plus the hashTreeRoot of the SyncCommittee is usually already available due to state hashing.

**Comment by shresthagrawal** [comment]
@dapplion yes that works, I already mentioned that in the GitHub issue earlier https://github.com/ChainSafe/lodestar/issues/4733#issuecomment-1311457596

**Comment by dadepo** [comment]
> > Maybe consider adding tests that covers the added endpoint? apart from that LGTM.
> 
> it already covered the end point as long as we update the testData for lightclient, doubled checked it via the unit test output
> 
> ```
>   beacon / lightclient
> onError: Error: aborted
>     ✔ getStateProof
>     ✔ getUpdates
>     ✔ getOptimisticUpdate
>     ✔ getFinalityUpdate
>     ✔ getBootstrap
>     ✔ getCommitteeHash
> ```

Unfortunately that test only seem to test some of the type machinery and not the actual implementation. 

I updated the implementation to throw an error like this:

```
    async getCommitteeHash(startPeriod: SyncPeriod, count: number) {
      throw new Error("error");
      const maxAllowedCount = Math.min(MAX_REQUEST_LIGHT_CLIENT_COMMITTEE_HASHES, count);
      const periods = Array.from({length: maxAllowedCount}, (_ignored, i) => i + startPeriod);
      const committeeHashes = await Promise.all(
        periods.map((period) => chain.lightClientServer.getCommitteeHash(period))
      );
      return {data: committeeHashes};
    },
```

Ran the test and also got:

```
  beacon / lightclient
    ✔ getStateProof
    ✔ getUpdates
    ✔ getOptimisticUpdate
    ✔ getFinalityUpdate
    ✔ getBootstrap
    ✔ getCommitteeHash
```

You can also confirm this. 

What will be needed is e2e test, and the few existing ones can be found in `packages/beacon-node/test/e2e/api/impl/beacon`

**Comment by twoeths** [comment]
> What will be needed is e2e test, and the few existing ones can be found in packages/beacon-node/test/e2e/api/impl/beacon

@dadepo thanks, implemented it

File: packages/beacon-node/src/chain/lightClient/index.ts [hunk 3]
```diff
@@ -291,6 +293,31 @@ export class LightClientServer {
...
+    }
+
+    const currentSyncCommittee = await this.db.syncCommittee.get(syncCommitteeWitness.currentSyncCommitteeRoot);
+    if (!currentSyncCommittee) {
+      throw new LightClientServerError(
+        {code: LightClientServerErrorCode.RESOURCE_UNAVAILABLE},
+        `currentSyncCommittee not available ${toHexString(blockRoot)} for period ${period}`
+      );
+    }
+    return digest(concat(currentSyncCommittee.pubkeys));
```
Comment (dapplion) [inline packages/beacon-node/src/chain/lightClient/index.ts]: Do we agree on doing hashTreeRoot here or not?

Comment (twoeths) [reply packages/beacon-node/src/chain/lightClient/index.ts]: I can switch to use root unless @shresthagrawal wants to use hash for now

Comment (shresthagrawal) [reply packages/beacon-node/src/chain/lightClient/index.ts]: Yes please go ahead with the root!

**Review by wemeetagain — APPROVED** [review]
LGTM

**Pull request merged**